src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,26 @@
 import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from '../../app/store';
+import { setQuery, setStatus } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const { status, query } = useSelector((state: RootState) => state.filter);
+  const dispatch: AppDispatch = useDispatch();
+
+  const handleStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setStatus(event.target.value as Status));
```

Comment:
The `setStatus` action expects a value of type `Status`. Ensure that the `event.target.value` is a valid `Status` type. Consider validating or casting it properly.